### PR TITLE
[MIG] [16.0] base_graphql

### PR DESCRIPTION
**Pull request by rconjour**
[MIG] [16.0] base_graphql

### CODE REVIEW

**Comment by paulius-sladkevicius** [comment]
As I'm keen to see it in the v16, could be worth to ping @sbidoul :-)

File: graphql_base/controllers/main.py [hunk 1]
```diff
@@ -77,7 +61,7 @@ def _process_request(self, schema, data):
             return response
         except HttpQueryError as e:
             result = json_encode({"errors": [{"message": str(e)}]})
-            headers = dict(e.headers)
+            headers = dict()
```
Comment (sbidoul) [inline graphql_base/controllers/main.py]: Why this change?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -10,7 +10,6 @@ exclude: |
   ^base_rest_pydantic/|
   ^datamodel/|
   ^extendable/|
-  ^graphql_base/|
```
Comment (sbidoul) [inline .pre-commit-config.yaml:13]: Could you do graphql_demo in this same PR, or apply https://github.com/OCA/maintainer-tools/wiki/Use-temporary-reference%28s%29-to-another-pull-request%28s%29 in #324 so we can test on runboat?

**Review by sbidoul — COMMENTED** [review]
Thanks for working on this @rconjour !

**Comment by sbidoul** [comment]
Hm, dear bot, why did you not notify me about this PR, since I'm a maintainer of the migrated module?

File: graphql_base/controllers/main.py (hunk 1 above)
Comment (rconjour) [reply graphql_base/controllers/main.py]: The HttpQueryError does not contain any headers anymore

File: .pre-commit-config.yaml (hunk 2 above)
Comment (rconjour) [reply .pre-commit-config.yaml:13]: Done!

File: graphql_base/controllers/main.py (hunk 1 above)
Comment (sbidoul) [reply graphql_base/controllers/main.py]: I still see it [here](https://github.com/graphql-python/graphql-server/blob/master/graphql_server/error.py#L18), though. And this does not seem to have changed recently.

Comment (sbidoul) [inline graphql_base/controllers/main.py]: Hm, it looks like `e.headers` can be `None`, so I suggest this:
```suggestion
            headers = dict(e.headers or {})
```

Comment (rconjour) [reply graphql_base/controllers/main.py]: Done!

**Comment by sbidoul** [comment]
Thanks!

/ocabot merge patch
